### ISSUE TITLE
NodeJSRuntime crashes when JSON object is used to reject the returned Promise

### ISSUE THREAD

**Issue by mdeuser** [openwhisk/openwhisk#1351]
## Environment details:

* local deployment, vagrant, ubuntu VM, Windows   

## Steps to reproduce the issue:

1.   Copy the following JS snippet into test.js
```
function main(message) {
    var badReject = { "field1": "field 1 value"};
    var goodReject = JSON.stringify(badReject);

    if (message.ret == 'bad') {
        retVal = badReject;
    } else {
        retVal = goodReject;
    }

    return new Promise(function (resolve, reject) {
       reject(retVal);
    });
}
```
2.   `wsk action create test test.js`
3.   `wsk action invoke test -p ret bad --blocking` 

The next steps work correctly, but may be helpful in troubleshooting
  [4].   `wsk action invoke test -p ret bad`    ### NOTE:  this is an async invocation
  [5].   `wsk activation get ACTIVATIONID`

## Provide the expected results and outputs:
Blocking invocations should return a properly formatted response
`wsk action invoke test -p ret bad --blocking`
```
...
    "response": {
        "status": "application error",
        "success": false,
        "result": {
            "error": {
                "field1": "field 1 value"
            }
        }
    },
```
Non-blocking invocations' activation record should contain a properly formatted response
`wsk action invoke test -p ret bad`
```
ok: invoked /_/test with id ccfc8d0ffc664afc907e32b22f9d6450
```
`wsk activation get ccfc8d0ffc664afc907e32b22f9d6450`
```
...
    "response": {
        "status": "application error",
        "success": false,
        "result": {
            "error": {
                "field1": "field 1 value"
            }
        }
    },
```

## Provide the actual results and outputs:
`wsk action invoke test -p ret bad --blocking`
```
error: Unable to invoke action 'test': The connection failed, or timed out. (HTTP status code 502)
```
`wsk action invoke test -p ret bad`
```
ok: invoked /_/test with id ccfc8d0ffc664afc907e32b22f9d6450
> wsk activation get ccfc8d0ffc664afc907e32b22f9d6450
...
    "response": {
        "status": "application error",
        "success": false,
        "result": {
            "error": {
                "field1": "field 1 value"
            }
        }
    },
```

## Additional information you deem important:
* It seems that something in the use of --blocking causes the reject payload to crash the invoker runtime, which results in a connection failure.

**Comment by jasonpet** [comment]
@mdeuser - You had mentioned a duplicate issue that James Dubee was looking into.  Can you please reference that issue here?

**Comment by mdeuser** [comment]
@jasonpet - Sure.  It's #1394

**Comment by jasonpet** [comment]
this was resolved by #1394
